### ISSUE TITLE
`cf.Field.regrids` doesn't seem to behave properly while regridding from rotated latitude-longitude to regular latitude-longitude?

### ISSUE THREAD

**Issue by bewithankit** [NCAS-CMS/cf-python#571]
### The version of the software and the environment:
```python
>>> cf.environment(paths=False)
Platform: Linux-3.10.0-1160.76.1.el7.x86_64-x86_64-with-glibc2.17
HDF5 library: 1.12.1
netcdf library: 4.8.1
udunits2 library: /opt/jaspy/lib/libudunits2.so.0
Python: 3.10.5
netCDF4: 1.6.0
cftime: 1.6.1
numpy: 1.22.4
psutil: 5.9.1
scipy: 1.8.1
matplotlib: 3.5.2
ESMF: 8.2.0
cfdm: 1.9.0.4
cfunits: 3.3.4
cfplot: 3.1.23
cf: 3.13.0
```

```python
>>> %matplotlib inline
>>> import cf
>>> import cfplot as cfp
>>> import numpy as np

>>> f = cf.read('/home/users/ankitkat/test/au952a.pd20510414.pp')
print(f)
[<CF Field: id%UM_m01s03i463_vn1006(time(8), grid_latitude(432), grid_longitude(444))>]

>>> gust = f[0]
>>> gust.dump()
-----------------------------------------------------------
Field: id%UM_m01s03i463_vn1006 (ncvar%UM_m01s03i463_vn1006)
-----------------------------------------------------------
Conventions = 'CF-1.9'
_FillValue = -1073741824.0
history = 'Converted from UM/PP by cf-python v3.13.0'
lbproc = '8192'
lbtim = '122'
long_name = 'WIND GUST'
runid = 'aaaaa'
source = 'UM vn1006'
stash_code = '3463'
submodel = '1'
um_stash_source = 'm01s03i463'

Data(time(8), grid_latitude(432), grid_longitude(444)) = [[[5.587890625, ..., 5.1376953125]]]

Cell Method: time(8): maximum

Domain Axis: grid_latitude(432)
Domain Axis: grid_longitude(444)
Domain Axis: height(1)
Domain Axis: time(8)

Dimension coordinate: time
    axis = 'T'
    calendar = '360_day'
    standard_name = 'time'
    units = 'days since 2051-1-1'
    Data(time(8)) = [2051-04-14 01:30:00, ..., 2051-04-14 22:30:00] 360_day
    Bounds:calendar = '360_day'
    Bounds:units = 'days since 2051-1-1'
    Bounds:Data(time(8), 2) = [[2051-04-14 00:00:00, ..., 2051-04-15 00:00:00]] 360_day

Dimension coordinate: height
    axis = 'Z'
    positive = 'up'
    standard_name = 'height'
    units = 'm'
    Data(height(1)) = [-1.0] m

Dimension coordinate: grid_latitude
    axis = 'Y'
    standard_name = 'grid_latitude'
    units = 'degrees'
    Data(grid_latitude(432)) = [-24.474999085068703, ..., 22.93500065803528] degrees
    Bounds:units = 'degrees'
    Bounds:Data(grid_latitude(432), 2) = [[-24.52999908477068, ..., 22.990000657737255]] degrees

Dimension coordinate: grid_longitude
    axis = 'X'
    standard_name = 'grid_longitude'
    units = 'degrees'
    Data(grid_longitude(444)) = [-29.47499145567417, ..., 19.255008280277252] degrees
    Bounds:units = 'degrees'
    Bounds:Data(grid_longitude(444), 2) = [[-29.52999145537615, ..., 19.31000827997923]] degrees

Auxiliary coordinate: latitude
    standard_name = 'latitude'
    units = 'degrees_north'
    Data(grid_latitude(432), grid_longitude(444)) = [[20.576467692711244, ..., 66.9022518505943]] degrees_north
    Bounds:units = 'degrees_north'
    Bounds:Data(grid_latitude(432), grid_longitude(444), 4) = [[[20.505853650744182, ..., 66.82752183591477]]] degrees_north

Auxiliary coordinate: longitude
    standard_name = 'longitude'
    units = 'degrees_east'
    Data(grid_latitude(432), grid_longitude(444)) = [[-10.577446822867152, ..., 68.72895292160312]] degrees_east
    Bounds:units = 'degrees_east'
    Bounds:Data(grid_latitude(432), grid_longitude(444), 4) = [[[-10.602339269012656, ..., 68.73573608505069]]] degrees_east

Coordinate reference: grid_mapping_name:rotated_latitude_longitude
    Coordinate conversion:grid_mapping_name = rotated_latitude_longitude
    Coordinate conversion:grid_north_pole_latitude = 39.25
    Coordinate conversion:grid_north_pole_longitude = 198.0

>>> print(gust.coordinate('time').datetime_array)
[cftime.Datetime360Day(2051, 4, 14, 1, 30, 0, 0, has_year_zero=True)
 cftime.Datetime360Day(2051, 4, 14, 4, 30, 0, 0, has_year_zero=True)
 cftime.Datetime360Day(2051, 4, 14, 7, 30, 0, 0, has_year_zero=True)
 cftime.Datetime360Day(2051, 4, 14, 10, 30, 0, 0, has_year_zero=True)
 cftime.Datetime360Day(2051, 4, 14, 13, 30, 0, 0, has_year_zero=True)
 cftime.Datetime360Day(2051, 4, 14, 16, 30, 0, 0, has_year_zero=True)
 cftime.Datetime360Day(2051, 4, 14, 19, 30, 0, 0, has_year_zero=True)
 cftime.Datetime360Day(2051, 4, 14, 22, 30, 0, 0, has_year_zero=True)]

>>> test = cf.dt(2051, 4, 14, 1, 30, 0, 0)
>>> print(test)
2051-04-14 01:30:00

>>> cfp.mapset(resolution='50m')
>>> cfp.con(gust.subspace(T=test), lines=False)
```
![image](https://user-images.githubusercontent.com/83757805/214858263-4e44d660-f5b8-4ae0-8a83-f0954513e5c4.png)
```python
>>> cfp.mapset(resolution='50m', proj='rotated')
>>> cfp.con(gust.subspace(T=test), lines=False)
```
![image](https://user-images.githubusercontent.com/83757805/214859006-a8f304c8-c7d7-4815-9d6b-035e2af767ef.png)
### Creating dimension coordinates for the destination grid and performing spherical regridding:
```python
>>> rotated_latitude = gust.coord('grid_latitude').array
>>> rotated_longitude = gust.coord('grid_longitude').array

>>> target_latitude = cf.DimensionCoordinate(data=cf.Data(np.linspace(-90, 90, num=180), 'degrees_north'))
>>> target_longitude = cf.DimensionCoordinate(data=cf.Data(np.linspace(-180, 180, num=360), 'degrees_east'))

>>> lon_bounds = target_longitude.create_bounds()
>>> lat_bounds = target_latitude.create_bounds()

>>> target_longitude.set_bounds(lon_bounds)
>>> target_latitude.set_bounds(lat_bounds)

>>> regridded_data = gust.regrids({'latitude': target_latitude, 'longitude': target_longitude}, 'linear')
>>> regridded_data.dump()
-----------------------------------------------------------
Field: id%UM_m01s03i463_vn1006 (ncvar%UM_m01s03i463_vn1006)
-----------------------------------------------------------
Conventions = 'CF-1.9'
_FillValue = -1073741824.0
history = 'Converted from UM/PP by cf-python v3.13.0'
lbproc = '8192'
lbtim = '122'
long_name = 'WIND GUST'
runid = 'aaaaa'
source = 'UM vn1006'
stash_code = '3463'
submodel = '1'
um_stash_source = 'm01s03i463'

Data(time(8), latitude(180), longitude(360)) = [[[--, ..., --]]]

Cell Method: time(8): maximum

Domain Axis: height(1)
Domain Axis: latitude(180)
Domain Axis: longitude(360)
Domain Axis: time(8)

Dimension coordinate: time
    axis = 'T'
    calendar = '360_day'
    standard_name = 'time'
    units = 'days since 2051-1-1'
    Data(time(8)) = [2051-04-14 01:30:00, ..., 2051-04-14 22:30:00] 360_day
    Bounds:calendar = '360_day'
    Bounds:units = 'days since 2051-1-1'
    Bounds:Data(time(8), 2) = [[2051-04-14 00:00:00, ..., 2051-04-15 00:00:00]] 360_day

Dimension coordinate: height
    axis = 'Z'
    positive = 'up'
    standard_name = 'height'
    units = 'm'
    Data(height(1)) = [-1.0] m

Dimension coordinate: longitude
    standard_name = 'longitude'
    units = 'degrees_east'
    Data(longitude(360)) = [-180.0, ..., 180.0] degrees_east
    Bounds:units = 'degrees_east'
    Bounds:Data(longitude(360), 2) = [[-180.50139275766017, ..., 180.50139275766017]] degrees_east

Dimension coordinate: latitude
    standard_name = 'latitude'
    units = 'degrees_north'
    Data(latitude(180)) = [-90.0, ..., 90.0] degrees_north
    Bounds:units = 'degrees_north'
    Bounds:Data(latitude(180), 2) = [[-90.50279329608938, ..., 90.50279329608938]] degrees_north

Coordinate reference: grid_mapping_name:rotated_latitude_longitude
    Coordinate conversion:grid_mapping_name = rotated_latitude_longitude
    Coordinate conversion:grid_north_pole_latitude = 39.25
    Coordinate conversion:grid_north_pole_longitude = 198.0
```
### Plotting the `regridded_data` then gives output which is not expected:
```python
>>> cfp.mapset(resolution='50m')
>>> cfp.con(regridded_data.subspace(T=test), lines=False)
```
![image](https://user-images.githubusercontent.com/83757805/214862399-c9c7db0d-d1ae-4e82-beb0-052b1b449af5.png)

### I then tried exploring the `cf.Field.constructs` for the `regridded_data`:
```python
>>> print(regridded_data.constructs)
Constructs:
{'cellmethod0': <CF CellMethod: domainaxis0: maximum>,
 'coordinatereference0': <CF CoordinateReference: grid_mapping_name:rotated_latitude_longitude>,
 'dimensioncoordinate0': <CF DimensionCoordinate: time(8) days since 2051-1-1 360_day>,
 'dimensioncoordinate1': <CF DimensionCoordinate: height(1) m>,
 'dimensioncoordinate2': <CF DimensionCoordinate: longitude(360) degrees_east>,
 'dimensioncoordinate3': <CF DimensionCoordinate: latitude(180) degrees_north>,
 'domainaxis0': <CF DomainAxis: size(8)>,
 'domainaxis1': <CF DomainAxis: size(1)>,
 'domainaxis2': <CF DomainAxis: size(180)>,
 'domainaxis3': <CF DomainAxis: size(360)>}
```
### Thinking `grid_mapping_name:rotated_latitude_longitude` to be the potential culprit I thought of deleting it altogether:
```python
>>> regridded_data.del_construct('coordinatereference0')
<CF CoordinateReference: grid_mapping_name:rotated_latitude_longitude>
```
### Plotting the `regridded_data` then gives output which is expected:
```python
>>> cfp.mapset(resolution='50m')
>>> cfp.con(regridded_data.subspace(T=test), lines=False)
```
![image](https://user-images.githubusercontent.com/83757805/214867758-9feca0f8-42f1-4ed3-991b-81d352fecdc0.png)

> **I don't know if the preservation of  `grid_mapping_name:rotated_latitude_longitude` after regridding from a rotated latitude-longitude source domain to a regular latitude-longitude destination domain is a bug or my approach is wrong. Thank you.**

**Comment by davidhassell** [comment]
Hi Ankit, thanks for working through this. The problem arises due to an error in the reading of the PP file - the rotated pole coordinate reference construct is lacking the mappings to its coordinates. It _should_ have looked like:
```python
Coordinate reference: grid_mapping_name:rotated_latitude_longitude
    Coordinate conversion:grid_mapping_name = rotated_latitude_longitude
    Coordinate conversion:grid_north_pole_latitude = 39.25
    Coordinate conversion:grid_north_pole_longitude = 198.0
    Dimension coordinate: grid_latitude
    Dimension coordinate: grid_longitude
    Auxiliary coordinate: latitude
    Auxiliary coordinate: longitude
```

This is fixed in version 3.14.0, out on 2023-01-31, but the work-around for now is just as you have done, i.e. deleting the offending coordinate reference construct.

**Comment by bewithankit** [comment]
Thank you David!
